proxy/http/HttpSM.cc

```diff
@@ -6502,6 +6504,8 @@ HttpSM::setup_100_continue_transfer()
   // Make sure the half_close is not set.
   ua_txn->set_half_close_flag(false);
   ua_entry->in_tunnel = true;
+  ink_assert(ua_entry->vc != nullptr);
+  ua_entry->vc->make_tunnel_endpoint();
```

Comment:
`ua_entry->vc` is a `VConnection` pointer, which is why `make_tunnel_endpoint()` has to be virtual member function of `VConnection`.